rasa/cli/run.py

```diff
@@ -75,28 +75,54 @@ def _validate_model_path(model_path: Text, parameter: Text, default: Text):
 def run(args: argparse.Namespace):
     import rasa.run
 
-    args.model = _validate_model_path(args.model, "model", DEFAULT_MODELS_PATH)
-
-    if not args.enable_api:
-        # if the API is enabled you can start without a model as you can train a
-        # model via the API once the server is up and running
-        from rasa.model import get_model
-
-        try:
-            get_model(args.model)
-        except ModelNotFound:
-            print_error(
-                "No model found. Train a model before running the "
-                "server using `rasa train` and use '--model' to provide "
-                "the model path."
-            )
-            return
-
     args.endpoints = get_validated_path(
         args.endpoints, "endpoints", DEFAULT_ENDPOINTS_PATH, True
     )
     args.credentials = get_validated_path(
         args.credentials, "credentials", DEFAULT_CREDENTIALS_PATH, True
     )
 
-    rasa.run(**vars(args))
+    if args.enable_api:
+        args.model = _validate_model_path(args.model, "model", DEFAULT_MODELS_PATH)
+        rasa.run(**vars(args))
+
+    # if the API is not enable you cannot start without a model
+    # make sure either a model server, a remote storage, or a local model is
+    # configured
+
+    from rasa.model import get_model
+    from rasa.core.utils import AvailableEndpoints
+
+    # start server if remote storage is configured
+    if args.remote_storage is not None:
+        rasa.run(**vars(args))
+
+    # start server if model server is configured
```

Comment:
Last question: Is this order still correct now?
```
1. Fetch the model from a server (see :ref:`server_fetch_from_server`), or
2. Fetch the model from a server (see :ref:`server_fetch_from_server`), or	2. Fetch the model from a remote storage (see :ref:`cloud-storage`).
3. Fetch the model from a remote storage (see :ref:`cloud-storage`).	3. Load the model specified via ``-m`` from your local storage system,
```

Reply:
yes

Reply:
If `args.remote_storage is not None` are you going to make it to the model server?

Reply:
there order here does not really matter as the server should just start as long as one of them is set

Reply:
so if a remote storage is set, it does not mean that in the end this one is also taken. if a model server is configured that would be used in the end. however, the checks we are doing here is just about making sure that at lease some kind of model configuration exists as you are not able to train a model afterwards if the api is not enabled.

Reply:
Ah okay I gotcha, thanks for the explanation!